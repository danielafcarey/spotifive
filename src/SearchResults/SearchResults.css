.search-results p {
  text-align: center; }
