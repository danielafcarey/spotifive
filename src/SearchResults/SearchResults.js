import React from 'react';
import { string, func, arrayOf, shape, object } from 'prop-types';
import Instructions from '../Instructions/Instructions';
import ArtistCard from '../ArtistCard/ArtistCard';

const SearchResults = (props) => {
  const { selectArtist } = props;
  const { searchResults } = props.searchResults;

  if (searchResults.length < 1) {
    return <Instructions />;
  } else {
    const listOfResults = searchResults.map((result, index) => {
      return (
        <ArtistCard 
          name={ result.name } 
          id={ result.id }
          image={ result.image }
          selectArtist={ selectArtist } 
          key={`artistKey${index}`}
        />     
      );
    });

    return (
      <div className='search-results'>
        <p>Select an artist: </p>
        <ul>
          { listOfResults }
        </ul>
      </div>
    );
  }
};

SearchResults.propTypes = {
  selectArtist: func,
  searchResults: shape({
    searchResults: arrayOf(object),
    searchError: string
  })
};

export default SearchResults;
