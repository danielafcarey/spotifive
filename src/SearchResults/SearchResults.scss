.search-results {
  ul {
    padding: 0 20px;
  }

  p {
    text-align: center;
  }
}
