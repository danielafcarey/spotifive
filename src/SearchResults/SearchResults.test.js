import React from 'react';
import { shallow } from 'enzyme';
import SearchResults from './SearchResults';

describe('SearchResults', () => {
  let mockProps;
  
  beforeEach(() => {
    mockProps = {
      selectArtist: jest.fn(),
      searchResults: {
        searchResults: [],
        searchError: null
      }
    }
  })

  it('matches the snapshot if there are no search results', () => {
    const wrapper = shallow(<SearchResults { ...mockProps } />)

    expect(wrapper).toMatchSnapshot();
  })

  it('matches the snapshot if there are search results', () => {
    mockProps.searchResults.searchResults = ['result'];
    const wrapper = shallow(<SearchResults { ...mockProps } />);

    expect(wrapper).toMatchSnapshot();
  })
})
