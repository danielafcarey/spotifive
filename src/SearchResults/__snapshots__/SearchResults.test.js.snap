// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResults matches the snapshot if there are no search results 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchResults
    searchResults={
        Object {
            "searchError": null,
            "searchResults": Array [],
          }
    }
    selectArtist={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {},
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`SearchResults matches the snapshot if there are search results 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchResults
    searchResults={
        Object {
            "searchError": null,
            "searchResults": Array [
              "result",
            ],
          }
    }
    selectArtist={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p>
          Select an artist: 
</p>,
        <ul>
          <ArtistCard
                    id={undefined}
                    image={undefined}
                    name={undefined}
                    selectArtist={[Function]}
          />
</ul>,
      ],
      "className": "search-results",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Select an artist: ",
        },
        "ref": null,
        "rendered": "Select an artist: ",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <ArtistCard
              id={undefined}
              image={undefined}
              name={undefined}
              selectArtist={[Function]}
/>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "artistKey0",
            "nodeType": "function",
            "props": Object {
              "id": undefined,
              "image": undefined,
              "name": undefined,
              "selectArtist": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "ul",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p>
            Select an artist: 
</p>,
          <ul>
            <ArtistCard
                        id={undefined}
                        image={undefined}
                        name={undefined}
                        selectArtist={[Function]}
            />
</ul>,
        ],
        "className": "search-results",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Select an artist: ",
          },
          "ref": null,
          "rendered": "Select an artist: ",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <ArtistCard
                id={undefined}
                image={undefined}
                name={undefined}
                selectArtist={[Function]}
/>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "artistKey0",
              "nodeType": "function",
              "props": Object {
                "id": undefined,
                "image": undefined,
                "name": undefined,
                "selectArtist": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "ul",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
